twoliter/embedded/partyplanner

```diff
@@ -247,27 +234,51 @@ set_partition_sizes() {
   pp_size["BIOS"]="${BIOS_MIB}"
   ((offset += BIOS_MIB))
 
-  for bank in A B; do
-    pp_offset["EFI-${bank}"]="${offset}"
-    pp_size["EFI-${bank}"]="${EFI_MIB}"
-    ((offset += EFI_MIB))
+  if [[ "${in_place_updates}" == "yes" ]]; then
+    # Allocate two partition banks to support in-place updates.
+    for bank in A B; do
+      pp_offset["EFI-${bank}"]="${offset}"
+      pp_size["EFI-${bank}"]="${EFI_MIB}"
+      ((offset += EFI_MIB))
 
-    pp_offset["BOOT-${bank}"]="${offset}"
-    pp_size["BOOT-${bank}"]="${boot_mib}"
-    ((offset += boot_mib))
+      pp_offset["BOOT-${bank}"]="${offset}"
+      pp_size["BOOT-${bank}"]="${boot_mib}"
+      ((offset += boot_mib))
 
-    pp_offset["ROOT-${bank}"]="${offset}"
-    pp_size["ROOT-${bank}"]="${root_mib}"
-    ((offset += root_mib))
+      pp_offset["ROOT-${bank}"]="${offset}"
+      pp_size["ROOT-${bank}"]="${root_mib}"
+      ((offset += root_mib))
 
-    pp_offset["HASH-${bank}"]="${offset}"
-    pp_size["HASH-${bank}"]="${hash_mib}"
-    ((offset += hash_mib))
+      pp_offset["HASH-${bank}"]="${offset}"
+      pp_size["HASH-${bank}"]="${hash_mib}"
+      ((offset += hash_mib))
 
-    pp_offset["RESERVED-${bank}"]="${offset}"
-    pp_size["RESERVED-${bank}"]="${reserved_mib}"
-    ((offset += reserved_mib))
-  done
+      pp_offset["RESERVED-${bank}"]="${offset}"
+      pp_size["RESERVED-${bank}"]="${reserved_mib}"
+      ((offset += reserved_mib))
+    done
+  else
+    # Allocate a single partition bank, where each partition is twice as large.
+    pp_offset["EFI-A"]="${offset}"
+    pp_size["EFI-A"]="$(( EFI_MIB * 2 ))"
+    ((offset += EFI_MIB * 2))
+
+    pp_offset["BOOT-A"]="${offset}"
+    pp_size["BOOT-A"]="$(( boot_mib * 2 ))"
+    ((offset += boot_mib * 2))
+
+    pp_offset["ROOT-A"]="${offset}"
+    pp_size["ROOT-A"]="$(( root_mib * 2 ))"
+    ((offset += root_mib * 2))
+
+    pp_offset["HASH-A"]="${offset}"
+    pp_size["HASH-A"]="$(( hash_mib * 2 ))"
```

Comment:
To clarify my understanding, this is just to keep things easy to track by making double but we really don't need nor will we use twice the HASH-A size, correct?

Reply:
> To clarify my understanding, this is just to keep things easy to track by making double but we really don't need nor will we use twice the HASH-A size, correct?

The size of the verity tree increases with the size of the block device data. It's not a linear increase the way this implies, though, so this is mostly for consistency.